File: java/two-fer/src/main/java/Twofer.java
class Twofer {

    private static final String ANSWER_TEMPLATE = "One for %s, one for me.";
    private static final String NO_NAME_VALUE = "you";

    String twofer(String name) {

        String twoferAnswer = null;

        if (name == null || name.trim().isEmpty()) {
            twoferAnswer = String.format(ANSWER_TEMPLATE, NO_NAME_VALUE);
        }
        else{

            twoferAnswer = String.format(ANSWER_TEMPLATE, name);
        }

        return twoferAnswer;
    }
}

File: go/two-fer/two_fer.go
// Package twofer allows to create a 2-fer message: One for you and one for me.
package twofer

import "fmt"

const defaultName = "you"

// ShareWith receives a name and returns the message "One for X, one for me".
// The X is "you" when name is missing.
func ShareWith(name string) string {
	if name == "" {
		name = defaultName
	}
	return fmt.Sprintf("One for %s, one for me.", name)
}
